### DISCUSSION TITLE
Incorporate OBS features

### DISCUSSION THREAD

**Discussion by e-Feitosa** [sircharlo/meeting-media-manager#1325] [Ideas] [answered]
Looking at some details of OBS Studio, and being an open source program as well. Actually it's a doubt / and a speculation: Is it possible to think about incorporating two resources in the M3 so as not to depend on OBS Studio?
1. Virtual camera
2. Create scenes

Why this question?
Some cannot use the M3, because it (m³) uses OBS Studio to transition scenes.

I know that this would involve many hours of typing and that perhaps it would not be feasible to think of an improvement project like this, but I am curious if you already had the idea of incorporating these resources in M3 so as not to depend on OBS Studio. This would help a lot in setting up the M3 and make it easier for many who depend on a machine (PC or Laptop) a little better to run OBS Studio + M3.

**Comment by mtdvlpr** [comment, marked as answer]
Hi @e-Feitosa,

> Is it possible to think about incorporating two resources in the M3 so as not to depend on OBS Studio?

I don't know, but like you said, it would require a great deal of research and effort.

> Some cannot use the M3, because it (m³) uses OBS Studio to transition scenes.

Could you explain? OBS is an optional integration, so M³ can be used without it.

> This would help a lot in setting up the M3 and make it easier [...]

Could you elaborate? I don't see the benefit. You end up with the same features, now possible in 2 different ways.

**Reply by e-Feitosa** [reply]
I'll try to explain better. I know it's possible to just use M3, but as far as I understand, option 1 is to use M3 without streaming for the zoom meeting, and option 2 is to use OBS Studio to transition images/videos to KH and zoom meeting simultaneously, that is, this happens in hybrid meetings, so we use the camera via OBS Studio (virtualcam) to make the transition, and, in OBS Studio, we create the camera scene and the media scene, and we can also create other scenes (stage, speaker, table etc). It is in this second option that I asked about this optional configuration that we do in OBS Studio, if it would be possible to include it in the M3 configurations.

**hybrid meetings**
⬇️
![image](https://user-images.githubusercontent.com/90984283/215351204-16002b73-004b-4c89-939f-eadcadd156a0.png)

_[Currently, we use OBS Studio to capture the M3 image (MEDIA SCENE) and transition between the Main Scene and the media.]_

**Reply by mtdvlpr** [reply]
@e-Feitosa, alright, let's see if I get this right. You want to continue to use OBS, but you want to be able to modify the OBS scenes from within M³ which would change the scenes set in OBS for you?

**Reply by e-Feitosa** [reply]
Almost like that, using the M3, without OBS Studio, that is, the M3 would receive the camera signal and create a virtual camera for the zoom meeting. Above is how it currently works, the idea would be to eliminate the use of OBS Studio, incorporating the virtual camera option in the M3 and creating scenes.
Today my OBS Studio looks like this:
[The camera and the Scene (or Scenes) would only be configured in the M3 and not in OBS Studio.]
![image](https://user-images.githubusercontent.com/90984283/215351094-fdac45c6-f2fe-4062-9d26-95c9378a37a6.png)

**Reply by mtdvlpr** [reply]
@e-Feitosa, got it. In my opinion this is out of scope.
1. The features are already present using OBS
2. If using OBS is too much, those features inside M3 would probably also be too much
3. It would cost a lot of time and effort (if it's even feasible) that could be used for new features instead

Maybe @sircharlo disagrees, but I doubt it.

**Reply by e-Feitosa** [reply]
I'm glad you paid attention to this post! As I mentioned earlier, it's a single point that I notice that some have difficulty setting up OBS Studio. But without a doubt, the M3 program is excellent every day in everything, without errors, without bugs, and working perfectly with or without the optional feature by OBS Studio. Congratulations on the work you two do (@sircharlo and @mtdvlpr).

**Reply by mtdvlpr** [reply]
@e-Feitosa glad to hear it! What we could do is improve the documentation about the OBS feature and how to set it up. I'll add it to the list :)

**Comment by vendelocity** [comment]
We love M3! After we update the week's media, we use a custom python script to create all the scenes and sources in OBS studio. This works great. Perhaps if just creating the scenes was an optional feature, this would be enough.

**Reply by e-Feitosa** [reply]
Hello, I don't understand why you always need to create new scenes in OBS Studio.

**Reply by vendelocity** [reply]
We create scenes for use, the same way you would create a playlist in VLC player. This way the program is always in a linear format that any attendant can follow.

**Reply by mtdvlpr** [reply]
@vendelocity I don't understand. You only need one media scene in OBS to display every media item. The source should be the M3 yeartext window or the second monitor in the Kingdom Hall. This doesn't need to change. Any additional scenes would be for different camera angles, like a stage view, speaker view, demonstration view, etc.

**Reply by e-Feitosa** [reply]
@mtdvlpr, I also don't understand why he needs to create the scenes. It may be that he is not creating the MEDIA SCENE in OBS Studio to present the media.

@vendelocity, My OBS Studio looks like this:
![image](https://github.com/sircharlo/meeting-media-manager/assets/90984283/0ec3db50-db11-4f64-b25d-6cd251b7d2ee)
